src/PoolManager.sol

```diff
@@ -174,8 +177,13 @@ contract PoolManager is IPoolManager, Fees, NoDelegateCall, Claims {
     }
 
     modifier onlyByLocker() {
+        // todo fix stack too deep :/
         address locker = Lockers.getCurrentLocker();
-        if (msg.sender != locker) revert LockedBy(locker);
+        if (msg.sender != locker) {
+            if (msg.sender != CurrentHookAddress.get() || !Hooks.shouldAccessLock(IHooks(CurrentHookAddress.get()))) {
```

Comment:
Cache `CurrentHookAddress.get()`?

Reply:
I get stack too deep :/ Any ideas? I left a comment above to trim but I couldn't think of anything

Reply:
btw was able to fix stack too deep by doing this

https://github.com/Uniswap/v4-core/compare/access-lock...stack-too-deep?expand=1 

but it adds gas :/